src/hooks/resources/application.ts

```diff
@@ -149,19 +149,73 @@ sbvrUtils.addPureHook('DELETE', 'resin', 'application', {
 				}
 			}
 			if (appIds.length > 0) {
-				// We need to null `should_be_running__release` or otherwise we have a circular dependency and cannot delete either
+				// find devices which are
+				// not part of any of the applications that are about to be deleted
+				// but run a release that belongs to any of the applications that
+				// is about to be deleted
 				return args.api
-					.patch({
-						resource: 'application',
+					.get({
+						resource: 'device',
+						passthrough: {
+							req: root,
+						},
 						options: {
+							$select: ['uuid'],
 							$filter: {
-								id: { $in: appIds },
-								should_be_running__release: { $ne: null },
+								belongs_to__application: {
+									$not: {
+										$in: appIds,
+									},
+								},
+								is_running__release: {
```

Comment:
You need to `$or` `should_be_running__release` as well, right?

Reply:
`should_be_running__release` is cleared when a device is moved.